packages/phone-number-privacy/signer/test/integration/pnp.test.ts

```diff
@@ -225,33 +264,544 @@ describe('pnp', () => {
+            throw new Error('dummy error')
+          })
+          const req = getPnpQuotaRequest(ACCOUNT_ADDRESS1)
+          const authorization = getPnpRequestAuthorization(req, PRIVATE_KEY1)
+          const res = await sendRequest(req, authorization, SignerEndpoint.PNP_QUOTA)
+
+          expect(res.status).toBe(500)
+          expect(res.body).toMatchObject<PnpQuotaResponseFailure>({
+            success: false,
+            version: expectedVersion,
+            error: ErrorMessage.FAILURE_TO_GET_TOTAL_QUOTA,
+          })
+        })
+      })
     })
   })
 
-  // const sendPnpSignatureRequest = async (
-  //   req: SignMessageRequest,
-  //   authorization: string,
-  //   signerApp: any = app
-  // ) => {
-  //   return request(signerApp)
-  //     .get(SignerEndpoint.PNP_SIGN)
-  //     .set('Authorization', authorization)
-  //     .send(req)
-  // }
-  // // TODO: add signature tests
-  // describe(`${SignerEndpoint.PNP_SIGN}`, () => {
-  //   // it('Should return 200 and correct version', async () => {
-  //   //   const res = await request(app).get(SignerEndpoint.STATUS)
-  //   //   expect(res.status).toBe(200)
-  //   //   expect(res.body.version).toBe(expectedVersion)
-  //   // })
-  // })
+  describe(`${SignerEndpoint.PNP_SIGN}`, () => {
+    describe('quota calculation logic', () => {
+      quotaCalculationTestCases.forEach(({ expectedTotalQuota, cusdOdisPaymentInWei }) => {
+        it(`Should get totalQuota=${expectedTotalQuota} 
+           for cUSD (wei) payment=${cusdOdisPaymentInWei.toString()}`, async () => {
+          mockOdisPaymentsTotalPaidCUSD.mockReturnValue(cusdOdisPaymentInWei)
+
+          const req = getPnpSignRequest(
+            ACCOUNT_ADDRESS1,
+            BLINDED_PHONE_NUMBER,
+            AuthenticationMethod.WALLET_KEY,
+            IDENTIFIER
+          )
+          const authorization = getPnpRequestAuthorization(req, PRIVATE_KEY1)
+          const res = await sendRequest(req, authorization, SignerEndpoint.PNP_SIGN)
+
+          const shouldSucceed = expectedTotalQuota > 0
+
+          if (shouldSucceed) {
+            expect(res.status).toBe(200)
+            expect(res.body).toMatchObject<SignMessageResponseSuccess>({
+              success: true,
+              version: expectedVersion,
+              signature: expectedSignature,
+              performedQueryCount: 1, // incremented for signature request
+              totalQuota: expectedTotalQuota,
+              blockNumber: testBlockNumber,
+              warnings: [],
+            })
+          } else {
+            expect(res.status).toBe(403)
+            expect(res.body).toMatchObject<SignMessageResponseFailure>({
+              success: false,
+              version: expectedVersion,
+              performedQueryCount: 0,
+              totalQuota: expectedTotalQuota,
+              blockNumber: testBlockNumber,
+              error: WarningMessage.EXCEEDED_QUOTA,
+            })
+          }
+        })
```

Comment:
could you also add cases for:
- performedQuota == totalQuota (non-zero)
- performedQuota > totalQuota

Reply:
I'll add these in the "endpoint functionality" section below